client/homebrew/editor/metadataEditor/metadataEditor.jsx

```diff
@@ -51,11 +56,33 @@ const MetadataEditor = createClass({
 	},
 
 	handleFieldChange : function(name, e){
-		this.props.onChange({
-			...this.props.metadata,
-			[name] : e.target.value
+		e.persist();
+		const inputRules = validations[name] ?? [];
+		const validationErr = inputRules.map((rule)=>rule(e.target.value)).filter(Boolean);
+		this.setState((prevState)=>({
+		    errs : {
+				...(prevState.errs ?? {}),
+				[name] : validationErr.length > 0 ? validationErr : undefined
+			}
+		}), ()=>{
```

Comment:
I'm not sure we need to save the errors to the state.

Reply:
correct, removed